src/AppInstallerCLICore/Workflows/DependenciesFlow.cpp

```diff
@@ -133,6 +135,104 @@ namespace AppInstaller::CLI::Workflow
+
+            rootDependencies.ApplyToType(DependencyType::WindowsFeature, [&hr, &dismHelper, &info, &warn, &force, &rebootRequired](Dependency dependency)
+                {
+                    if (SUCCEEDED(hr) || force)
+                    {
+                        auto featureName = dependency.Id();
+                        DismHelper::WindowsFeature windowsFeature = dismHelper.CreateWindowsFeature(featureName);
+
+                        if (windowsFeature.DoesExist())
+                        {
+                            if (!windowsFeature.IsEnabled())
+                            {
+                                std::string featureDisplayName = AppInstaller::Utility::ConvertToUTF8(windowsFeature.GetDisplayName());
+                                Utility::LocIndView locIndFeatureName{ featureName };
+
+                                info << Resource::String::EnablingWindowsFeature(Utility::LocIndView{ featureDisplayName }, locIndFeatureName) << std::endl;
+
+                                hr = windowsFeature.Enable();
+                                AICLI_LOG(Core, Info, << "Enabling Windows Feature [" << featureName << "] returned with HRESULT: " << hr);
+
+                                if (FAILED(hr))
+                                {
+                                    warn << Resource::String::FailedToEnableWindowsFeature(locIndFeatureName) << std::endl;
+                                }
+
+                                if (hr == ERROR_SUCCESS_REBOOT_REQUIRED || windowsFeature.GetRestartRequiredStatus() == DismRestartType::DismRestartRequired)
+                                {
+                                    rebootRequired = true;
+                                }
+                            }
+                        }
+                        else
+                        {
+                            // Note: If a feature is not found, continue enabling the rest of the dependencies but block immediately after unless force arg is present.
+                            AICLI_LOG(Core, Info, << "Windows Feature [" << featureName << "] does not exist");
+                            hr = APPINSTALLER_CLI_ERROR_INSTALL_MISSING_DEPENDENCY;
+                            warn << Resource::String::WindowsFeatureNotFound(Utility::LocIndView{ featureName }) << std::endl;
+                        }
+                    }
+            });
+
+            if (FAILED(hr) || hr == APPINSTALLER_CLI_ERROR_INSTALL_MISSING_DEPENDENCY)
+            {
+                if (force)
+                {
+                    context.Reporter.Warn() << Resource::String::FailedToEnableWindowsFeatureOverridden << std::endl;
+                }
+                else
+                {
+                    context.Reporter.Error() << Resource::String::FailedToEnableWindowsFeatureOverrideRequired << std::endl;
+                    AICLI_TERMINATE_CONTEXT(hr);
+                }
+            }
+            else if (rebootRequired)

```

Comment:
I think this neglects the case where one feature fails to enable, and another has a reboot required. I don't know if we care about that case or not, or if that's even a possibility with the way `hr` gets handled.

If `--force` isn't present, I don't think it matters, since both the reboot and the fail would cause the context to terminate, and the termination reason doesn't make much of a difference. If `--force` is present, the user would only be informed they are bypassing the failed install, and they wouldn't be informed they are also bypassing the reboot requirement.